src/lib/update-clients.ts

```diff
@@ -0,0 +1,46 @@
+import axios from "axios";
+import { GET_OAUTH_CLIENT, PUT_OAUTH_CLIENT } from "./constants";
+import { OAuthAppConfigAxios } from "./update-oauth-client";
+
+export interface OAuthClientConfig {
+  urls: string[];
+  name: string;
+}
+interface ClientParams {
+  clients: OAuthClientConfig[];
+  AM_URL: string;
+  ssoToken: string;
+  cookieName: string;
+  realm: string;
+}
+async function updateClients({
+  AM_URL,
+  cookieName,
+  realm,
+  clients,
+  ssoToken,
+}: ClientParams): Promise<void> {
+  const request = axios.create({
+    baseURL: AM_URL,
+    headers: {
+      "Content-Type": "application/json",
+      "Accept-API-Version": "protocol=2.0,resource=1.0",
+      [cookieName]: ssoToken,
+    },
+  });
+
+  for (const { urls: signinUrl, name } of clients) {
+    const amConfigUrl = GET_OAUTH_CLIENT.url(realm, name);
+    try {
+      const { data: config }: { data: OAuthAppConfigAxios } = await request.get(
+        amConfigUrl
+      );
+      config.coreOAuth2ClientConfig.redirectionUris.value.push(...signinUrl);
```

Comment:
Do we want to conditionally check for property existence prior to diving that deep, or are you just catching any thrown errors that may happen from this?

Reply:
This isnt correct yet.